app/main/authorize/ayr_user.py

```diff
@@ -0,0 +1,68 @@
+from typing import List
+
+from sqlalchemy import exc
+
+from app.main.authorize.keycloak_manager import (
+    get_user_transferring_body_keycloak_groups,
+)
+from app.main.db.models import Body, db
+
+
+class AYRUser:
+    def __init__(self, groups: List[str]) -> None:
+        self.groups = groups
+
+    @property
+    def can_access_ayr(self):
+        return self.is_superuser or self.is_standard_user
+
+    @property
+    def is_superuser(self) -> bool:
+        return "/ayr_user_type/view_all" in self.groups
+
+    @property
+    def is_standard_user(self) -> bool:
+        return "/ayr_user_type/view_dept" in self.groups
+
+    @property
+    def transferring_bodies(self) -> str | None:
+        if self.is_superuser or not self.is_standard_user:
+            return None
+
+        return get_user_accessible_transferring_bodies(self.groups)
+
+
+def get_user_accessible_transferring_bodies(groups: List[str]) -> List[str]:
```

Comment:
Was thinking of refactoring this a bit more later to make it that the AYRUser doesnt have to know about the db but will leave for now.